api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala

```diff
@@ -23,9 +23,13 @@ import za.co.absa.hyperdrive.ingestor.api.manager.StreamManager
   * Base class for all StreamDecoders.
   *
   */
-abstract class StreamWriter(destination: String) {
+abstract class StreamWriter {
 
   def write(dataFrame: DataFrame, streamManager: StreamManager): StreamingQuery
 
-  def getDestination: String = destination
+  /**
+   *
+   * @return a string describing the destination, used for logging
+   */
+  def getDestination: String
```

Comment:
Hmm I am not sure about this method here. It is not "functional" stuff used only as description. I should rather provide a default implementation and return child class here.

Reply:
It's the same as `StreamReader.getSourceName`. Maybe I could rename it to `getDestinationName` to clarify that it's only a name and has no functional usage.

Reply:
I know that ... but I am against "no functional" aspect of this. Each implementation should log its own stuff. 
For example there is code ` logger.info(s"STARTING ingestion from '${streamReader.getSourceName}' into '${streamWriter.getDestination}' (id = $ingestionId)")
`, information that Ingestion was started is fine but source and destination log should be done in component which is responsible for that.

Reply:
I leave it up to you. I am a fine event with your impl.

Reply:
I'm going to change it. I also don't like it